### PR TITLE
2973 asset list view

### LINKED ISSUE #2973: Asset list view
Part of epic #2896 

## Is your feature request related to a problem? Please describe 👀
[Figma designs](https://www.figma.com/file/rIK2zVFUVhtbuXo0RBzOzw/0033-2305_omSupply-ColdChain-UI?type=design&node-id=2653-16424&mode=design&t=8KRUuAo7s1V0r0RM-0)

## Describe the solution you'd like 🎁
image

my notes 
* the phrase 'New CCE' isn't grammatically correct and perhaps there is a better option. If we used `Asset` it wouldn't be a problem, but the intention here is to create a subclass of asset which is strictly cold chain equipment
* The scan icon isn't standard
* does 'export' save to csv as per other list views?
* differing layout depending on permission : if you can access multi facilities, then shown facility as a column
* add pagination to the table

image

Requirements doc has 

**Permissions:** 
1. Only users with the permissions “View Asset Register” can see the assets
2. Only user with the permissions “Edit Asset Register” can edit the assets details
3. Store-level users can only view/edit assets assigned to their store

**Navigation:** 
1. Home > Cold Chain menu > Equipment sub menu >  Cold Chain Equipment Inventory: List View (for assets with class = Cold Chain Equipment)
2. … > Asset Catalogue:  Detailed View > Click on “View Register” > Asset/CCE Register: List View 

**Headers:** 
1. Breadcrumbs: Assets / Breadcrumbs: Equipment
2. Filters
3. Buttons:
  1. Add Asset / Add CCE
  2. Import button
  3. Export button
  4. Scan QR Code
4. Select [Action] dropdown: 
5. Delete selected lines

**Description: to create / edit Asset Register / CCE Inventory**
1. Export the asset register / CCE Inventory to add new assets / CCE or to update existing records
5. Import the new/updated asset register / CCE Inventory

**Columns**
* Manufacturer
* Model
* Code
* Description
* Serial number
* Store / customer
* Store name's category 1 (level 2) e.g. province (note: we don't have this in omSupply)
* Store name's category 1 (level 3) e.g. district  (note: we don't have this in omSupply)
* Inventory storage location(s) (multiple values in this text field)
* Functional status
 

### Moneyworks Jobcode 🧰

`OMSCC:AST`

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #2973 #2975

# 👩🏻‍💻 What does this PR do? 
 
Does the basic list view for Asset / CCE items. 
Some of the items will be held over for a new issue:

* Import spreadsheet
* Scan assets
* Batch delete (pending API implementation)
* Manufacturer and Model columns (pending PR merge for the loaders)

So, what's in this version?

* New menu item under cold chain
* List view page showing a list of assets
* Export of asset list to csv
* New asset creation : the modal has name and code added to the original mock up as the API requires these fields. seemed neater to add them into the modal
* filters
* sorting
* pagination

Screenshot 2024-03-01 at 4 39 32 PM

Screenshot 2024-03-01 at 4 37 27 PM

# 🧪 How has/should this change been tested? 

You will need to create some assets, either through the UI or directly into the database.
Check that filter and sort work and that the assets display correctly.

## 📃 Documentation

- [ ] Asset list

### CODE REVIEW

**Comment by jmbrunskill** [comment]
There's some kind of state problem I think, Trying to create two assets in a row gives an AssetAlreadyExists error.
I think the id may be being re-used?
image

**Review by jmbrunskill — APPROVED** [review]
I'm happy if you want to merge, but I think should either implement the Loader stuff as it's now merged, or create a follow up issue.

Assets get created with no `store_id`, is that what we want?
image

I'm thinking maybe want to restrict creating an owned assets to the central server?
Or at least we need to think about how we can assign a store...
Do we have anything in the Update UI for this?

File: client/packages/coldchain/src/Equipment/ListView/ListView.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const navigate = useNavigate();
+  const t = useTranslation('catalogue');
+  const { localisedDate } = useFormatDateTime();
+  const modalController = useToggle();
+
+  const columns = useColumns<AssetFragment>(
+    [
+      ['code', { width: 150, sortable: false }],
+      // {
+      //   key: 'manufacturer',
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply client/packages/coldchain/src/Equipment/ListView/ListView.tsx:39]: yeah, but the loaders didn't exist when I did it.

File: client/packages/coldchain/src/Equipment/ListView/CreateAssetModal.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+} from '@openmsupply-client/system';
+import { useAssets } from '../api';
+
+interface CreateAssetModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const mapCatalogueItem = (catalogueItem: AssetCatalogueItemFragment) => ({
+  label: `${catalogueItem.code} Type Manufacturer Model`,
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply client/packages/coldchain/src/Equipment/ListView/CreateAssetModal.tsx:29]: no loaders :wink

File: client/packages/coldchain/src/Equipment/ListView/Toolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import { mapIdNameToOptions, useAssetData } from '@openmsupply-client/system';
+
+type ReferenceData = {
+  id: string;
+  name: string;
+  categoryId?: string;
+};
+
+export const Toolbar = () => {
+  const { data: categoryData } = useAssetData.utils.categories();
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply client/packages/coldchain/src/Equipment/ListView/Toolbar.tsx:18]: we do need to, I'll add a filter

File: client/packages/coldchain/src/Equipment/ListView/ListView.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply client/packages/coldchain/src/Equipment/ListView/ListView.tsx:39]: image

**Pull request merged**